exonum/src/blockchain/block.rs

```diff
@@ -207,6 +210,103 @@ impl BlockProof {
             non_exhaustive: (),
         }
     }
+
+    /// Verifies that the block in this proof is endorsed by the Byzantine majority of provided
+    /// validators.
+    pub fn verify(&self, validator_keys: &[PublicKey]) -> Result<(), ProofError> {
+        if self.precommits.len() < byzantine_quorum(validator_keys.len()) {
+            return Err(ProofError::NoQuorum);
+        }
+        if self.precommits.len() > validator_keys.len() {
+            return Err(ProofError::DoubleEndorsement);
+        }
+
+        let correct_heights = self
+            .precommits
+            .iter()
+            .all(|precommit| precommit.payload().height == self.block.height);
+        if !correct_heights {
```

Comment:
Does it make sense to recognise which one of the precommits has a wrong height ?

Reply:
I don't see a clear use case for this, tbh.